src/main/java/hwr/oop/riddler/model/Sudoku.java

```diff
@@ -1,14 +1,67 @@
 package hwr.oop.riddler.model;
 
 import hwr.oop.riddler.model.component.Box;
+import hwr.oop.riddler.model.component.Cell;
 import hwr.oop.riddler.model.component.Column;
 import hwr.oop.riddler.model.component.Row;
 
+import java.util.ArrayList;
+
 public class Sudoku {
+    public Cell[][] cells = new Cell[9][9];
```

Comment:
Das würde ich dann garnicht mehr als 2 dimensionales array speichern, da man sonst an 2 Stellen in der Sudoku Klasse die Struktur des Sudokos speicherst, also einmal ist die durch Rows und Columns gegeben, und dann nochmal durch das 2 dimensionale Array. 

+ es wäre wahrscheinlich auch einfacher damit zu arbeiten wenn cells ein einfaches Array oder ein Set ist.

Reply:
Alternativ könnte man aber auch rows und columns nicht als klassenatribut haben, sondern immer dynamisch generieren, sobald man z.B. die getRows Methode aufruft.

Reply:
Ja, vielleicht macht es sinn die CellGroup Objekte dynamisch zu generieren, da wir sie sonst updaten müssen,

Reply:
Die Rows und Columns sind grade nur dazu da um die werte in der CellGroup zu speichern. Sie sind also nur für die Struktur des Sudokus geeignet. Vielleicht sollten wir sie lieber RowSet, ColumnSet, und BoxSet nennen. Was meinst du?

Reply:
Dynamisch generieren klingt gut, RowSet würde auch passen, vielleicht aber doch einfach die Reihenfolge dann im dynamisch generierten mit zurückgeben? Solange man das nirgendwo speichert hat man das Problem ja eigentlich auch nicht.